### PR TITLE
create new unit modal

### PR DESCRIPTION
**Pull request by GaganSinghGit**
This PR is part of UI enhancements team for OnTrack. This is the component review for create-unit-modal.

### CODE REVIEW

**Comment by maddernd** [comment]
Please can you add a description for what this is as part of  your PR, Also it currently has [EOF Rule] File "docs/OnTrack/UI Enhancement/ComponentReviewCreateUnitModal.md" has lint errors and this will need to be resolved ( Line Lint errors)

**Comment by GaganSinghGit** [comment]
Hi how do I resolve the lint error? I am using vs-code. I am assuming there is an extension I can use to format it as per requirements. Let me know.

**Comment by maddernd** [comment]
Its a formatting error, just google the best way to solve it.

Can  you also add some screenshots of your component that you want to work on to make the review clear.

**Comment by GaganSinghGit** [comment]
What specific/exact formatting error is it? How can I find that out?
I cannot see the error on my end. I believe I have to install a linter or code formatter to find that out, but I dont know which one are we using for this project.

**Comment by maddernd** [comment]
Run fernandrone/linelint@master
/usr/bin/docker run --name c044[2](https://github.com/thoth-tech/documentation/actions/runs/4461363098/jobs/7835139483?pr=191#step:4:3)b0d0e91f97e9[4](https://github.com/thoth-tech/documentation/actions/runs/4461363098/jobs/7835139483?pr=191#step:4:5)9f28a29a820a2ccf1ef_ee2[5](https://github.com/thoth-tech/documentation/actions/runs/4461363098/jobs/7835139483?pr=191#step:4:6)12 --label [6](https://github.com/thoth-tech/documentation/actions/runs/4461363098/jobs/7835139483?pr=191#step:4:7)c0442 --workdir /github/workspace --rm -e "HOME" -e "GITHUB_JOB" -e "GITHUB_REF" -e "GITHUB_SHA" -e "GITHUB_REPOSITORY" -e "GITHUB_REPOSITORY_OWNER" -e "GITHUB_REPOSITORY_OWNER_ID" -e "GITHUB_RUN_ID" -e "GITHUB_RUN_NUMBER" -e "GITHUB_RETENTION_DAYS" -e "GITHUB_RUN_ATTEMPT" -e "GITHUB_REPOSITORY_ID" -e "GITHUB_ACTOR_ID" -e "GITHUB_ACTOR" -e "GITHUB_TRIGGERING_ACTOR" -e "GITHUB_WORKFLOW" -e "GITHUB_HEAD_REF" -e "GITHUB_BASE_REF" -e "GITHUB_EVENT_NAME" -e "GITHUB_SERVER_URL" -e "GITHUB_API_URL" -e "GITHUB_GRAPHQL_URL" -e "GITHUB_REF_NAME" -e "GITHUB_REF_PROTECTED" -e "GITHUB_REF_TYPE" -e "GITHUB_WORKFLOW_REF" -e "GITHUB_WORKFLOW_SHA" -e "GITHUB_WORKSPACE" -e "GITHUB_ACTION" -e "GITHUB_EVENT_PATH" -e "GITHUB_ACTION_REPOSITORY" -e "GITHUB_ACTION_REF" -e "GITHUB_PATH" -e "GITHUB_ENV" -e "GITHUB_STEP_SUMMARY" -e "GITHUB_STATE" -e "GITHUB_OUTPUT" -e "RUNNER_OS" -e "RUNNER_ARCH" -e "RUNNER_NAME" -e "RUNNER_TOOL_CACHE" -e "RUNNER_TEMP" -e "RUNNER_WORKSPACE" -e "ACTIONS_RUNTIME_URL" -e "ACTIONS_RUNTIME_TOKEN" -e "ACTIONS_CACHE_URL" -e GITHUB_ACTIONS=true -e CI=true -v "/var/run/docker.sock":"/var/run/docker.sock" -v "/home/runner/work/_temp/_github_home":"/github/home" -v "/home/runner/work/_temp/_github_workflow":"/github/workflow" -v "/home/runner/work/_temp/_runner_file_commands":"/github/file_commands" -v "/home/runner/work/documentation/documentation":"/github/workspace" 6c0442:b0d0e91f97e949f28a29a820a2ccf1ef
No configuration file found at ./.linelint.yml (will use default configuration)
[EOF Rule] File "docs/OnTrack/UI Enhancement/ComponentReviewCreateUnitModal.md" has lint errors

Total of 1 lint errors!

**Comment by maddernd** [comment]
You can see it under the checks tab, I would advise you get two plugins Prettier - code formatter and markdownlint in VS code, they will help you solve these errors.

**Comment by GaganSinghGit** [comment]
Hi, linter and pretties errors are fixed. None of the vale errors are from my files

**Review by macite — COMMENTED** [review]
What is it going to look like when it is built? Please add a sketch.

File: docs/OnTrack/UI Enhancement/ComponentReviewCreateUnitModal.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
+# Component Review
+
+## Student Name: Gaganjeet Singh
+
+## Student ID: 220032936
+
+## Term: T1 2023 – SIT 378
+
+## Ontrack Component review check
```
Comment (macite) [inline docs/OnTrack/UI Enhancement/ComponentReviewCreateUnitModal.md:9]: Please remove this - it is good in the instructions but not needed here.

File: docs/OnTrack/UI Enhancement/ComponentReviewCreateUnitModal.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+## Component Name
+
+\*\*\*\*./src/app/admin/modals/create-unit-modal/create-unit-modal.coffee
+
+**File Name:**create-unit-modal.coffee
+
+## Component purpose
+
+## It is used to create a new unit
```
Comment (macite) [inline docs/OnTrack/UI Enhancement/ComponentReviewCreateUnitModal.md:23]: This is not a heading...

File: docs/OnTrack/UI Enhancement/ComponentReviewCreateUnitModal.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+## Component outcomes/interactions
+
+It creates a new unit. ![image](Images/CreateUnitModal.png)
```
Comment (macite) [inline docs/OnTrack/UI Enhancement/ComponentReviewCreateUnitModal.md:27]: With what details... I suggest adding a list here.

The user provides:
- unit code
- name
- teaching period or start and end dates

Then you could put in the new sketch - highlighting changes. eg use of material components, and existing components from the project (I think there is one to select a teaching period)

File: docs/OnTrack/UI Enhancement/ComponentReviewCreateUnitModal.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+**Component migration Check list** – What is needs to be checked for this component to work once
```
Comment (macite) [inline docs/OnTrack/UI Enhancement/ComponentReviewCreateUnitModal.md:29]: Rewrite the instruction here... eg: When working, this component needs to demonstrate the following:

then create a checklist:

[ ] ability to collect details from the user
[ ] succeeds when data is valid
[ ] handles errors - duplicate unit code in the teaching period, or invalid dates
[ ] created unit is shown on success

File: docs/OnTrack/UI Enhancement/ComponentReviewCreateUnitModal.md [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+1. The modal for creating the unit is working.
+
+   Existing API endpoint can be used
+
+2. Add should have a dropdown for teaching period.
+
+   Update API to accommodate this extra field
+
+3. Needs redesign (use material design)
```
Comment (macite) [inline docs/OnTrack/UI Enhancement/ComponentReviewCreateUnitModal.md:40]: We need a plan for the design now to make sure that it fits.

**Review by macite — CHANGES_REQUESTED** [review]
Please make changes as requested.

**Comment by maddernd** [comment]
Is there any progress on the requested changes from Andrew?

**Comment by GaganSinghGit** [comment]
> Is there any progress on the requested changes from Andrew?

Working on it

**Comment by GaganSinghGit** [comment]
Made the requested updates. Let me know if you need any more changes/additions.

**Comment by maddernd** [comment]
Hi , I have looked through and its almost there, If you see Andrews comments in this chat, I have closed off the ones complete, there are still 2 parts that need small adjustments.

**Comment by GaganSinghGit** [comment]
Regarding 'We need a plan for the design now to make sure that it fits', what do we need in the document? Is the sketch screen shot enough?

**Comment by maddernd** [comment]
I think Andrew wanted you to add a checklist too, if you have a look at his comments and reply to each one, that way we can get this merged in.

File: docs/OnTrack/UI Enhancement/ComponentReviewCreateUnitModal.md (hunk 1 above)
Comment (GaganSinghGit) [reply docs/OnTrack/UI Enhancement/ComponentReviewCreateUnitModal.md]: done

File: docs/OnTrack/UI Enhancement/ComponentReviewCreateUnitModal.md (hunk 2 above)
Comment (GaganSinghGit) [reply docs/OnTrack/UI Enhancement/ComponentReviewCreateUnitModal.md]: done

File: docs/OnTrack/UI Enhancement/ComponentReviewCreateUnitModal.md (hunk 5 above)
Comment (GaganSinghGit) [reply docs/OnTrack/UI Enhancement/ComponentReviewCreateUnitModal.md]: I have attached a screenshot of the sketch for this one. I am not sure what you mean by 'plan for the design'.

**Review by maddernd — APPROVED** [review]

**Review by maddernd — APPROVED** [review]

**Review by maddernd — APPROVED** [review]

**Review by maddernd — APPROVED** [review]

**Review by macite — APPROVED** [review]
This looks good - we will need the option to either have a custom date or a teaching period... but other than that this looks good. Default to teaching period.

**Pull request merged**